app/src/main/java/com/goldenraven/padawanwallet/ui/wallet/WalletRootScreen.kt

```diff
@@ -186,6 +186,18 @@ fun BalanceBox(
                         end.linkTo(parent.end)
                     }
             ) {
+                val isRefreshing by viewModel.isRefreshing.collectAsState()
+                val transition = updateTransition(targetState = isRefreshing, label = "PressTransition")
+                val rotationAngle by transition.animateFloat(
+                    transitionSpec = {
+                        repeatable(
+                            iterations = Int.MAX_VALUE,
+                            animation = tween(durationMillis = 2000, easing = LinearEasing)
+                        )
+                    }, label = ""
+                ) { pressed ->
+                    if (pressed) 360f else 0f
+                }
                 Button(
```

Comment:
Let's disable the button while the wallet is syncing by setting
```kotlin
enabled = !isRefreshing
```
on it. The only thing is that the default colours for the disabled button make it look weird and ugly. Do you mind also fixing up the colours a bit so that the button appears disabled but is more in line with the rest of the colour scheme?

Reply:
Now the button is enabled/disabled according to `isRefreshing` state.
I have also changed its color scheme.